tasks/eslint.md

```diff
@@ -0,0 +1,104 @@
+   to `['error', { requireStringLiterals: true }]` 
+5. [no-loop-func](http://eslint.org/docs/rules/no-loop-func) (1 error) 
+   ([airbnb](http://eslint.org/docs/rules/valid-typeof)) 
+   ([example PR](https://github.com/zulip/zulip/pull/2408))
+6. [new-cap](http://eslint.org/docs/rules/new-cap) (1 error) to be changed
+   to `'new-cap': ['error', { newIsCap: true, capIsNew: false, }]`
+   ([airbnb](https://github.com/airbnb/javascript#naming--PascalCase))
+7. [no-empty](http://eslint.org/docs/rules/no-empty) (2 errors)
+8. [space-before-blocks](http://eslint.org/docs/rules/space-before-blocks) (2 errors) (--fix) 
+   ([airbnb 1](https://github.com/airbnb/javascript#whitespace--before-blocks), 
+   [2](https://github.com/airbnb/javascript#functions--signature-spacing))
+9. [yoda](http://eslint.org/docs/rules/yoda) (3 errors) (--fix)
+10. [brace-style](http://eslint.org/docs/rules/brace-style) (7 errors) (--fix) 
+    to be changed to `['error', '1tbs', { allowSingleLine: true }]`
+    ([airbnb](https://github.com/airbnb/javascript#blocks--cuddled-elses))
+11. [keyword-spacing](http://eslint.org/docs/rules/keyword-spacing) (12 errors) (--fix)
+    ([airbnb](https://github.com/airbnb/javascript#whitespace--around-keywords)) to be changed to 
+
+    ```
+       'keyword-spacing': ['error', {
+          before: true,
+          after: true,
+          overrides: {
+            return: { after: true },
+            throw: { after: true },
+            case: { after: true }
+          }
+        }],
+    ```
+
+12. [one-var](http://eslint.org/docs/rules/one-var) (32 errors) to be changed to `['error', 'never']` 
+13. [no-else-return](http://eslint.org/docs/rules/no-else-return) (39 errors)
+14. [no-plusplus](http://eslint.org/docs/rules/no-plusplus) (40 errors)
+    ([airbnb](https://github.com/airbnb/javascript#variables--unary-increment-decrement))
+15. [no-shadow](http://eslint.org/docs/rules/no-shadow) (56 errors)
+16. [max-len](http://eslint.org/docs/rules/max-len) (78 errors)
+    ([airbnb](https://github.com/airbnb/javascript#whitespace--max-len)) to be changed to 
+
+    ```
+        'max-len': ['error', 100, 2, {
+          ignoreUrls: true,
+          ignoreComments: false,
+          ignoreRegExpLiterals: true,
+          ignoreStrings: true,
+          ignoreTemplateLiterals: true,
+        }],
+    ```
+
+17. [quote-props](http://eslint.org/docs/rules/quote-props) (201 errors) (--fix) to be changed
+    to `['error', 'as-needed', { keywords: false, unnecessary: true, numbers: false }]`
+    ([airbnb](https://github.com/airbnb/javascript#objects--quoted-props))
+18. [no-unused-vars](http://eslint.org/docs/rules/no-unused-vars) (221 errors) to be changed
+    to `['error', { vars: 'local', args: 'after-used' }]` 
+19. [no-underscore-dangle](http://eslint.org/docs/rules/no-underscore-dangle) (410 errors)
+    ([airbnb](https://github.com/airbnb/javascript#naming--leading-underscore))
```

Comment:
We use leading _, i.e. `_foo`, for real purposes (methods to be used only in tests, for example).  I would be happy to be trailing underscore, though, if we can do that.